kqueen_ui/blueprints/ui/views.py

```diff
@@ -724,6 +732,42 @@ def handle(self, cluster_id):
         return redirect(request.environ.get('HTTP_REFERER', url_for('ui.index')))
 
 
+class ClusterSetPolicy(KQueenView):
+    decorators = [login_required]
+    methods = ['POST']
+    validation_hint = 'uuid'
+
+    def handle(self, cluster_id):
+        redirect_url = request.environ.get('HTTP_REFERER', url_for('ui.index'))
+
+        cluster = self.kqueen_request('cluster', 'get', fnargs=(cluster_id,))
+        policy = cluster['metadata'].get('network_policy', 'none')
+        if 'node_count' not in cluster.get('metadata', {}):
+            engine = cluster.get('provisioner', {}).get('engine', '<unknown>')
+            flash('{} engine doesn\'t support network policy.'.format(
+                prettify_engine_name(engine)), 'warning')
+            return redirect(redirect_url)
+
+        if policy.get('provider') == 'CALICO':
+            if policy.get('enabled'):
```

Comment:
Maybe I'm missing something here, but it seems that you are disabling the network policy if its enabled in the metadata, why? Is it some sort of toggling mechanism, turn it off if its on and vice versa?

Reply:
Exactly, it's a toggling